code/modules/mob/living/carbon/human/human_defense.dm

```diff
@@ -47,6 +47,10 @@
 		if(mind.martial_art && !incapacitated(FALSE, TRUE) && mind.martial_art.can_use(src) && mind.martial_art.deflection_chance) //Some martial arts users can deflect projectiles!
 			if(prob(mind.martial_art.deflection_chance))
 				if((mobility_flags & MOBILITY_USE) && dna && !dna.check_mutation(HULK)) //But only if they're otherwise able to use items, and hulks can't do it
+					if(!isturf(loc)) //if we're inside something and still got hit
+						P.force_hit = TRUE //The thing we're in passed the bullet to us. Pass it back, and tell it to take the damage.

```

Comment:
This is pretty big assumption here. I'd prefer if this interaction was defined somewhere.

Reply:
Yeah, I'm not totally happy with it. Honestly, I had forgotten carp was a thing when I made the MK-I change. On the other hand, if people can directly hit the mech now, do I need this for balance anymore? People will be able to break the mech, and so it'll play out the same as it did in a MK-II or Firefighter.

It's probably a change for another PR, but carp users probably shouldn't get to shoot ranged mech weapons anyway.

Reply:
Alright, just make a comment here this is strictly intended for mechas for now.

Reply:
Done.